scripts/start-setupVanillaTweaks

```diff
@@ -12,73 +12,149 @@ set -e -o pipefail
-if [[ "$VANILLATWEAKS_FILE" ]]; then
-  if [ ! -f "$VANILLATWEAKS_FILE" ]; then
-    log "ERROR: given VANILLATWEAKS_FILE file does not exist"
+  log "DATAPACKS: ${DATAPACKS}, VT_VERSION: ${VT_VERSION}"
+  log $(curl -X POST -F "packs=${DATAPACKS}" -F "version=${VT_VERSION}" $VT_ZIPDATA_URL)
+  DOWNLOAD_URL=$(curl -X POST -F "packs=${DATAPACKS}" -F "version=${VT_VERSION}" $VT_ZIPDATA_URL | jq -r '.link // empty')
+  log "DOWNLOAD_URL: ${DOWNLOAD_URL}"
+  if [ ! "$DOWNLOAD_URL" ]; then
+    log "ERROR: unable to retrieve datapack DOWNLOAD_URL from vanillatweaks.net!"
     exit 2
   fi
 
-  PACKS=$(jq -jc '.packs' $VANILLATWEAKS_FILE)
-  if [ ! "$PACKS" ]; then
-    log "ERROR: unable to retrieve packs from $VANILLATWEAKS_FILE"
+  TEMPZIP=/tmp/vanillatweaks.zip
+  if ! get -o $TEMPZIP "https://vanillatweaks.net${DOWNLOAD_URL}"; then
+    log "ERROR: failed to download datapacks from ${DOWNLOAD_URL}"
     exit 2
   fi
 
-  VT_VERSION=$(jq -jc '.version' $VANILLATWEAKS_FILE)
-  if [ ! "$VT_VERSION" ]; then
-    log "ERROR: unable to retrieve version from $VANILLATWEAKS_FILE"
-    exit 2
+  mkdir -p "$DATAPACKS_DIR"
+  if ! unzip -o -d "$DATAPACKS_DIR" $TEMPZIP; then
+    log "ERROR: failed to unzip the datapacks ${DATAPACKS} from ${$TEMPZIP}"
   fi
-fi
 
-# Download and unzip packs
-if [[ "$PACKS" ]] && [[ "$VT_VERSION" ]]; then
-  VT_ZIPDATA_URL=https://vanillatweaks.net/assets/server/zipdatapacks.php
-  DOWNLOAD_URL=$(curl -X POST -F "packs=${PACKS}" -F "version=${VT_VERSION}" $VT_ZIPDATA_URL | jq -r '.link')
+  rm -f $TEMPZIP
+}
+
+# Download and move crafting tweaks
+downloadCraftingtweaks(){
+  VT_FILE=$1
+  CRAFTINGTWEAKS=$(jq -jc '.packs // empty' $VT_FILE)
+  if [ ! "$CRAFTINGTWEAKS" ]; then
+    log "ERROR: unable to retrieve crafting tweaks from ${VT_FILE}"
+    exit 2
+  fi
+  VT_ZIPDATA_URL="https://vanillatweaks.net/assets/server/zipcraftingtweaks.php"
+  
+  DOWNLOAD_URL=$(curl -X POST -F "packs=${CRAFTINGTWEAKS}" -F "version=${VT_VERSION}" $VT_ZIPDATA_URL | jq -r '.link // empty')
   if [ ! "$DOWNLOAD_URL" ]; then
-    log "ERROR: unable to retrieve DOWNLOAD_URL from vanillatweaks.net!"
+    log "ERROR: unable to retrieve crafting tweak DOWNLOAD_URL from vanillatweaks.net!"
     exit 2
   fi
 
-  TEMPZIP=/tmp/vanillatweaks.zip
-  if ! get -o $TEMPZIP "https://vanillatweaks.net${DOWNLOAD_URL}"; then 
-    log "ERROR: failed to download from ${DOWNLOAD_URL}"
+  mkdir -p "$DATAPACKS_DIR"
+  if ! get -o $DATAPACKS_DIR/crafting_tweaks.zip "https://vanillatweaks.net${DOWNLOAD_URL}"; then
+    log "ERROR: failed to download crafting tweaks from ${DOWNLOAD_URL}"
+    exit 2
+  fi
+}
+
+# Download resource packs
+downloadResourcepacks(){
+  VT_FILE=$1
+  RESOURCEPACKS=$(jq -jc '.packs // empty' $VT_FILE)
+  if [ ! "$RESOURCEPACKS" ]; then
+    log "ERROR: unable to retrieve resourcepacks from ${VT_FILE}"
     exit 2
   fi
+  VT_ZIPDATA_URL="https://vanillatweaks.net/assets/server/zipresourcepacks.php"
```

Comment:
There seems to be a lot of repetition across these three download functions. It looks like passing in the URL to POST and the final destination directory might allow for consolidating.

Reply:
...and unzip or not.

Reply:
Very slight difference in url, but you're right we can just pass the different suffix in.